### PR TITLE
Add baseID member variable and allow for passing into constructor

### PR DESCRIPTION
**Pull request by teelur**
I've been using this library, and it's great! Super nice interface for getting the broadcast data.

I noticed you are using a global variable outside of the library for getting the baseID. Since this is just a constant integer, we can pass this into the constructor where it is stored as a private member variable (better encapsulation!). This could allow for multiple instances of MegaCAN with different baseIDs.

I also noticed we were never calling initializeCAN() on the examples, so they weren't working. Updated them to call it during setup(). I also updated the examples to use the new constructor.

Changes:
- Added a `_baseID` private member variable & updated all references of baseID to use this rather than the global
- Added setters and getters for the new member variable.
- Added a new constructor that passes in baseID: `MegaCAN(const uint32 baseID)`
- Default constructor now takes the global and stores it in `_baseID`
- Updated Readme to suggest the new constructor (better OOP practices to pass this into an object). Old way of doing things is still usable for legacy code, but probably shouldn't encourage new code to use it.
- MegaCAN.h and MegaCAN.cpp had some random code blocks that used tabs instead of spaces. Fixed them to use spaces to be consistent with the rest of the files in the project.

Testing:
I ran the two broadcasting examples on my car and they function without any issues.

### CODE REVIEW

File: MegaCAN.h [hunk 1]
```diff
@@ -2,463 +2,473 @@
...
-	uint8_t gps_outstatus = 0; //GPS status byte (bit 0 = E/W)
-	int8_t gps_altk = 0; //GPS altitude km, km
-	uint16_t gps_altm = 0; //GPS altitude m, m/s
-	float gps_speed = 0; //GPS speed, m/s
-	float gps_course = 0; //GPS course, deg
-	
-	//offset 63
-	float generic_pid_duty1 = 0; //Generic close-loop duty 1, %
-	float generic_pid_duty2 = 0; //Generic close-loop duty 2, %
+  // offset 0
```
Comment (teelur) [inline MegaCAN.h:28]: This big section had tabs instead of spaces, so I updated to match rest of file

File: MegaCAN.h [hunk 2]
```diff
@@ -2,463 +2,473 @@
...
 
 typedef struct MegaCAN_message_t {
   struct {
-    uint32_t toTable = 0; // table on "To" device
-    uint32_t toID = 0; // "To" device ID
-    uint32_t fromID = 0; // "From" device ID
-    uint32_t msgType = 0; // Type of message, "1" is a request, "2" is a response
-    uint32_t toOffset = 0; // offset on "To" device
-  } core; //unpacked
+    uint32_t toTable = 0;    // table on "To" device
```
Comment (teelur) [inline MegaCAN.h:10]: Aligned these comments for readability

File: MegaCAN.cpp [hunk 3]
```diff
@@ -50,468 +54,468 @@ void MegaCAN::setMSresp(MegaCAN_message_t recMsg, MegaCAN_message_t &respMsg, ui
...
-			msg.gps_altm = (data[2] << 8) | data[3];
-			msg.gps_speed = ((data[4] << 8) | data[5]) / (float)10;
-			msg.gps_course = ((data[6] << 8) | data[7]) / (float)10;
-			break;
-		case 63:
-			msg.generic_pid_duty1 = data[0] * (float)392 / (float)1000;
-			msg.generic_pid_duty2 = data[1] * (float)392 / (float)1000;
-			break;
-	}
+  switch (id - _baseId) {
```
Comment (teelur) [inline MegaCAN.cpp:57]: More tab -> spaces fixes

**Comment by mantonakakis1** [comment]
Thanks for all this! I’m in the middle of a move at the moment but I should have some time to go through everything in the new few weeks.

**Comment by mantonakakis1** [comment]
Thank you again! Merging all of these.

**Pull request merged**